src/Mod/Arch/ArchRoof.py

```diff
@@ -42,241 +52,328 @@ def QT_TRANSLATE_NOOP(ctxt,txt):
 #  Roofs are build from a closed contour and a series of
 #  slopes.
 
-__title__="FreeCAD Roof"
+__title__  = "FreeCAD Roof"
 __author__ = "Yorik van Havre", "Jonathan Wiedemann"
-__url__ = "http://www.freecadweb.org"
+__url__    = "http://www.freecadweb.org"
+
+
+def adjust_list_len (lst, newLn, val):
```

Comment:
Do not leave a space before the parenthesis.

Also, in Python it is good practice to be verbose, meaning that very short variable names are discouraged, for example, `(old_list, new_length, value)` is better than `(lst, newLn, val)`. The first one is self-documenting, which means other programmers will be able to get the intention of the function faster without reading the entire function or getting cues from comments.

Also, remember that now Arch has coding guidelines (they are the same as Draft's), meaning that we should use `snake_case` for variables and avoid `camelCase` except for classes.

Reply:
Full compliance with the coding guidelines is not the goal of this update. I'll be happy to look at that aspect in a future iteration.

IMO the guidelines are also problematic:
indentation=4
number of indentations=6
name of function: 'DraftGeomUtils.findIntersection'=31
name of variable: 'perpendicular'=13
max. line length=78?
78-24-31-13=10 characters left.

Something has got to give here.

Reply:
Well, there are tricks to reduce the line length, encapsulating blocks of code into smaller subfunctions is the best solution in this case. The 80 character limit is a traditional limit, but it's always better to be pragmatic; I don't think there is a problem if it goes a bit beyond this, like 90 or 100; it's just good to try to be consistent.